### PR TITLE
Feature/firestore groceries

### PR DESCRIPTION
**Pull request by Geno22**
Added firestore support for the groceries list. This update is not reflected in the application for now.

### CODE REVIEW

File: app/src/androidTest/java/com/github/houseorganizer/houseorganizer/FirestoreShopItemUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        assertThat(oldQ, is(not(newQ)));
+        assertThat(newQ, is(9L));
+        testLocalItem.setQuantity(FirebaseTestsHelper.TEST_ITEM.getQuantity());
+    }
+
+    @Test
+    public void setUnitUpdatesFirestore() throws ExecutionException, InterruptedException {
+        DocumentReference docRef = testLocalItem.getItemDocRef();
+        Task<DocumentSnapshot> t = docRef.get();
+        Tasks.await(t);
```
Comment (Defibrilator) [inline app/src/androidTest/java/com/github/houseorganizer/houseorganizer/FirestoreShopItemUnitTest.java:101]: Those two lines are used a lot of times, you could extract them

**Review by Defibrilator — CHANGES_REQUESTED** [review]
Tests look good ! just a few changes to make it a bit clearer to read.

**Review by aindreias — APPROVED** [review]
Pretty comprehensive tests, and clean code. Thanks for the new feature!

File: .idea/misc.xml [hunk 2]
```diff
@@ -3,6 +3,7 @@
   <component name="DesignSurface">
     <option name="filePathToZoomLevelMap">
       <map>
+        <entry key="..\:/Users/flori/OneDrive/Documents/AAA_EPFL/BA6/Software DP/House-Organizer/app/src/main/res/layout/activity_login.xml" value="0.17831813576494426" />
```
[the remark this answers is not in the document]
Comment (Geno22) [reply .idea/misc.xml:6]: Yeah I am not going to include it ^^

File: app/src/androidTest/java/com/github/houseorganizer/houseorganizer/FirebaseTestsHelper.java [hunk 3]
```diff
@@ -179,6 +185,14 @@ protected static void createHouseholds() throws ExecutionException, InterruptedE
...
+    protected static void createTestShopList() throws ExecutionException, InterruptedException {
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
+        User owner = new DummyUser("test", TEST_USERS_EMAILS[0]);
+        ShopList shopList = new ShopList(owner, TEST_SHOPLIST_NAME);
+        shopList.addItem(TEST_ITEM);
+        FirestoreShopItem.storeShopList(shopList, db.collection("shop_lists"), TEST_SHOPLIST_NAME);
+    }
+
     /**
      * This method will create 8 users, 3 households and a task list
```
[the remark this answers is not in the document]
Comment (Geno22) [reply app/src/androidTest/java/com/github/houseorganizer/houseorganizer/FirebaseTestsHelper.java:197]: Can do !

File: app/src/androidTest/java/com/github/houseorganizer/houseorganizer/FirestoreShopItemUnitTest.java (hunk 1 above)
Comment (Geno22) [reply app/src/androidTest/java/com/github/houseorganizer/houseorganizer/FirestoreShopItemUnitTest.java:101]: Probably yes I can

File: app/src/androidTest/java/com/github/houseorganizer/houseorganizer/FirestoreShopItemUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        FirestoreShopItem.storeShopList(shopList, root, shopList.getListName());
+        DocumentReference doc = root.document("MyNewList");
+        Task<DocumentSnapshot> t0 = doc.get();
+        Tasks.await(t0);
+        doc.delete();
+        assertThat(t0.getResult().get("name"), is("MyNewList"));
+        assertThat(t0.getResult().get("owner"), is(FirebaseTestsHelper.TEST_USERS_EMAILS[0]));
+        assertThat(((ArrayList<User>)t0.getResult().get("authorized")).isEmpty(), is(true));
+    }
+}
```
[the remark this answers is not in the document]
Comment (Geno22) [reply app/src/androidTest/java/com/github/houseorganizer/houseorganizer/FirestoreShopItemUnitTest.java:155]: I don't think attributes are appropriate here as these values are used only here but I put the repeating one in a variable

**Review by Defibrilator — APPROVED** [review]

**Pull request merged**